src/main/java/seedu/address/logic/commands/AddAppointmentCommand.java

```diff
@@ -80,6 +82,9 @@ public CommandResult execute(Model model) throws CommandException {
         String name = targetDoctor.getName().toString();
         Appointment toAdd = new Appointment(description, dateTime, targetPatient, name);
 
+        if (targetPatient.hasAppointmentOnTimeslot(toAdd)) {
+            throw new CommandException(MESSAGE_SAME_APPOINTMENT_TIME);
+        }
         if (targetPatient.hasAppointment(toAdd)) {
```

Comment:
<img width="1148" alt="image" src="https://github.com/AY2324S1-CS2103T-W09-3/tp/assets/64828959/bd675659-683b-4f8d-894b-ab41d037120c">

If previous add appointment command was an error, successful command -> error message still stays for some reason

Reply:
need the steps to recreate the problem to get a better insight into whats causing the issue

Reply:
issue seems to be created because of timeslotnotfoundexception

Reply:
<img width="943" alt="Screenshot 2023-11-07 at 6 14 22 PM" src="https://github.com/AY2324S1-CS2103T-W09-3/tp/assets/64345206/55398fea-ff13-46c8-8b3b-8c3a7df56e2c">

so, similar to pin ren, i can successfully add a new appointment, but error is thrown in the highlighted 2 lines. @wj331 can you provide insights into what this does? and why is it throwing a TimeSlotNotFoundException